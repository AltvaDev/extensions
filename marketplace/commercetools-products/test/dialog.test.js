/* global global */

'use strict';

const React = require('react');
const renderer = require('react-test-renderer');

const { mockComponent } = require('./helpers.js');

jest.mock('@contentful/forma-36-react-components', () => ({
  Button: mockComponent('button')
}));

const { CommerceToolsDialog } = require('./../src/dialog.js');

describe('CommerceToolsDialog', function() {
  test('renders correctly', function() {
    const extension = {};
    const parameters = {
      projectKey: '',
      clientId: '',
      clientSecret: '',
      apiUri: '',
      authUri: '',
      locale: 'en'
    };

    global.CTPicker = class CTPicker {
      init() {
        return Promise.resolve();
      }

      show() {
        return Promise.resolve();
      }
    };

    const tree = renderer
      .create(
        <CommerceToolsDialog extension={extension} parameters={parameters} isSingle={false} />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();

    delete global.CTPicker;
  });
});
