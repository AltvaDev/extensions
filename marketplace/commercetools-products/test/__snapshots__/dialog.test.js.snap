// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommerceToolsDialog renders correctly 1`] = `
Array [
  <div
    className="picker-container"
  />,
  <div
    buttonType="positive"
    className="button"
    onClick={[Function]}
  >
    Insert selected Products
  </div>,
  <div
    buttonType="muted"
    className="button"
    onClick={[Function]}
  >
    Cancel
  </div>,
]
`;
